backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchConfiguration.java

```diff
@@ -37,6 +37,80 @@
 
 public class ElasticSearchConfiguration {
 
+    public enum HostScheme {
+        HTTP("http"),
+        HTTPS("https");
+
+        public static Optional<HostScheme> maybe(String schemeAsString) {
+            return Optional.ofNullable(schemeAsString)
+                .map(HostScheme::of);
+        }
+
+        public static HostScheme of(String schemeValue) {
+            return Arrays.stream(values())
+                .filter(hostScheme -> hostScheme.value.equalsIgnoreCase(schemeValue))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(
+                    String.format("Unknown HostScheme '%s'", schemeValue)));
+        }
+
+        private final String value;
+
+        HostScheme(String value) {
+            this.value = value;
+        }
+    }
+
+    public static class Credential {
+
+        public static Optional<Credential> maybe(String username, String password) {
+            if (username == null && password == null) {
```

Comment:
`||`?

Reply:
I prefer to throw when only one is not present